src/Host/Client/Impl/rtvs/R/grid.R

```diff
@@ -13,7 +13,28 @@ grid.format <- function(x) {
   sapply(format(x, trim = TRUE, justify = "none"), grid.trim);
 }
 
-grid.data <- function(x, rows, cols) {
+grid.sort.df <- function(x, rows, cols, df.sort.expression) {
+  if(missing(df.sort.expression)) {
+    x.df <- as.data.frame(x)[rows, cols];
+  } else {
+    x.df <- as.data.frame(x)
+    x.df <- x.df[eval(parse(text = df.sort.expression)),][rows, cols];
```

Comment:
This feels wrong. The expression is basically required to use `x`, so now that argument name is set in stone. I would rather avoid eval here altogether, and instead make the expression argument a function, to which `x` is passed as an argument. So instead of passing it as `"do.call(order, c(x.df['col_name1'], -x.df['col_name2'], ...))"`, you would pass `function(x) do.call(order, c(x.df['col_name1'], -x.df['col_name2'], ...))`.